webclient/src/components/ShareButton.vue

```diff
@@ -0,0 +1,59 @@
+<script setup lang="ts">
+import { copyCurrentLink } from "@/utils/common";
+import { ref } from "vue";
+
+const props = defineProps(["coords", "name"]);
+
+const browser_supports_share = "share" in navigator;
+const copied = ref(false);
+
+function shareLink() {
+  if (navigator.share) {
+    navigator.share({
+      title: props.name,
+      text: document.title,
+      url: window.location.href,
+    });
+  }
+}
+</script>
+
+<template>
+  <div class="link-popover">
+    <strong>{{ $t("view_view.header.external_link.open_in") }}</strong>
+    <a
+      class="btn"
+      target="_blank"
+      :href="
+        'https://www.openstreetmap.org/?mlat=' +
+        coords.lat +
+        '&mlon=' +
+        coords.lon +
+        '#map=17/' +
+        coords.lat +
+        '/' +
+        coords.lon +
+        '&layers=T'
+      "
+      >OpenStreetMap</a
+    ><br />
+    <a
+      class="btn"
+      target="_blank"
+      :href="'https://www.google.com/maps/search/?api=1&query=' + coords.lat + '%2C' + coords.lon"
+      >Google Maps</a
+    >
+    <a class="btn" :href="'geo:' + coords.lat + ',' + coords.lon">
+      {{ $t("view_view.header.external_link.other_app") }}
+    </a>
+    <strong>{{ $t("view_view.header.external_link.share") }}</strong>
+    <button class="btn" @click="shareLink" v-if="browser_supports_share">
+      {{ $t("view_view.header.external_link.share_link") }}
+    </button>
+    <button
+      class="btn"
+      @click="copyLink()"
+      v-html="copyCurrentLink(copied) ? $t('view_view.header.external_link.copied') : $t('view_view.header.copy_link')"
```

Comment:
This causes the current URL to be copied to the clipboard every time you go to a details page, because it runs on setup